run-tests

```diff
@@ -139,3 +139,33 @@ if [[ $fail -eq 1 ]] ; then
 else
 	echo -e "$MSG_PASSED"
 fi
+
+
+if [[ $EXPECT_5LVL -eq 1 ]] ; then
```

Comment:
I would cut this entire block so that `run-tests` does either 4-level tests or 5-level tests but not both.  (This file is going to get huge if we do both.)

The `BEGIN` and `END` assignment logic can happen once, up before the first mmap+memset+fork test.  Each test should be modified to run mmap+memset+fork with the `BEGIN` and `END` bounds, as well as the other arguments (like `--fork_child --map_private`) that make up that test.